CMakeLists.txt

```diff
@@ -102,6 +107,14 @@ file(COPY ${CMAKE_SOURCE_DIR}/translations/translations.qrc DESTINATION ${CMAKE_
 
 set_property(SOURCE ${CMAKE_BINARY_DIR}/generated/SettingsHelper.hpp ${CMAKE_BINARY_DIR}/generated/SettingsInfo.cpp PROPERTY SKIP_AUTOGEN ON)
 
+# Perform clang-tidy checks after we have build all the third_party dependencies.
+if(USE_CLANG_TIDY)
+	message(STATUS "Clang tidy checks will be performed")
+	
+	set(CMAKE_CXX_CLANG_TIDY 
+		"clang-tidy;-header-filter=^(?!.*\\bthird_party\\b).*$;-checks=-*,bugprone-*,cppcoreguidelines-*,modernize-*,performance-*,readability-*,-modernize-use-trailing-return-type,-bugprone-suspicious-include,-cppcoreguidelines-pro-bounds-array-to-pointer-decay,-cppcoreguidelines-avoid-magic-numbers,-readability-magic-numbers,-cppcoreguidelines-owning-memory,-readability-braces-around-statements,-cppcoreguidelines-macro-usage,-cppcoreguidelines-avoid-c-arrays,-modernize-avoid-c-arrays,-readability-implicit-bool-conversion,-readability-implicit-bool-conversion,-cppcoreguidelines-pro-type-static-cast-downcast,-cppcoreguidelines-special-member-functions,-cppcoreguidelines-avoid-non-const-global-variables;-warnings-as-errors=*")
```

Comment:
Every check that starts with `-name-of-check` is disabled.  Every check that starts with `name-of-checks-*` enables all checks with those category. Different checks are separated with `,`.

We treat all warning as error so that compilation fails when clangd tidy reports something.

We exclude every header that is inside `third_party`